modules/Core/pages/panel/users_edit.php

```diff
@@ -210,19 +210,42 @@
                     $errors[] = $language->get('admin', 'select_user_group');
                 }
             }
-        } else {
-            if (Input::get('action') == 'delete') {
-                if ($user_query->id > 1) {
-                    EventHandler::executeEvent('deleteUser', [
-                        'user_id' => $user_query->id,
-                        'username' => $user_query->username,
-                        'email_address' => $user_query->email
-                    ]);
 
-                    Session::flash('users_session', $language->get('admin', 'user_deleted'));
-                }
+        } else if ((Input::get('action') == 'delete') && $user_query->id > 1) {
+                EventHandler::executeEvent('deleteUser', [
+                    'user_id' => $user_query->id,
+                    'username' => $user_query->username,
+                    'email_address' => $user_query->email
+                ]);
 
+                Session::flash('users_session', $language->get('admin', 'user_deleted'));
                 Redirect::to(URL::build('/panel/users'));
+        } else if ((Input::get('action') == 'change_password') && $user_query->id > 1 && !$view_user->canViewStaffCP()) {
+            $validation = Validate::check($_POST, [
+                'password' => [
+                    Validate::REQUIRED => true,
+                    Validate::MIN => 6
+                ],
+                'password_again' => [
+                    Validate::MATCHES => 'password'
+                ]
+            ])->messages([
+                'password' => [
+                    Validate::REQUIRED => $language->get('user', 'password_required'),
+                    Validate::MIN => $language->get('user', 'password_minimum_6')
+                ],
+                'password_again' => $language->get('user', 'passwords_dont_match')
+            ]);
+
+            if ($validation->passed()) {
+                $password = Input::get('password');
+                $encrypted_password = password_hash($password, PASSWORD_BCRYPT, ['cost' => 13]);
+                $view_user->update([
+                    'password' => $encrypted_password
+                ]);
+                Session::flash('edit_user_success', $language->get('admin', 'user_password_changed_successfully'));
+            } else {
+                Session::flash('edit_user_error', implode('\n', $validation->errors()));
```

Comment:
Hmm, this feels really dirty. Not sure there is anything else we can do here though... Revamping how we handle success/error messages to simplify everything would be nice eventually:tm:

Reply:
I can't really think of a better way to do this currently